lib/src/num_ext.dart

```diff
@@ -0,0 +1,87 @@
+/// Int extensions
+extension IntExt on int {
+  /// Returns ordinal of [int]
+  /// ``` dart
+  /// print(1.ordinal); // 1st
+  /// print(22.ordinal); // 22nd
+  /// print(143.ordinal); // 143rd
+  /// print(0.ordinal); // 0th
+  /// print((-1).ordinal); // -1st
+  /// print(12.ordinal); // 12th
+  /// print(69.ordinal); // 69th
+  /// ```
+  String get ordinal {
+    if (this < 0) {
```

Comment:
It makes little sense to support negative numbers and turn them into ordinals. If it were necessary, it would suffice to convert it to positive before calling the function.

I would change the implementation into
```dart
String get ordinal {
    return switch (this % 100) {
      11 || 12 || 13 => '${this}th',
      _ => switch (this % 10) {
          1 => '${this}st',
          2 => '${this}nd',
          3 => '${this}rd',
          _ => '${this}th'
        }
    };
  }
```